data-prepper-plugins/s3-source/README.md

```diff
@@ -3,7 +3,94 @@
+are not compressed.
+
+```
+source:
+  s3:
+    notification_type: "sqs"
+    codec:
+      newline:
+    compression: none
+    sqs:
+      queue_url: "https://sqs.us-east-1.amazonaws.com/123456789012/MyQueue"
+    aws:
+      region: "us-east-1"
+      sts_role_arn: "arn:aws:iam::123456789012:role/Data-Prepper"
+```
+
+## Configuration Options
+
+All Duration values are a string that represents a duration. They support ISO_8601 notation string ("PT20.345S", "PT15M", etc.) as well as simple notation Strings for seconds ("60s") and milliseconds ("1500ms").
+
+
+* `notification_type` : Must be `sqs`.
+
+* `compression` (Optional) : The compression algorithm to apply. May be one of: `none`, `gzip`, or `automatic`. Defaults to `none`.
+
+* `codec` (Required) : The coded to apply. Must be either `newline` or `json`.
+
+* `sqs` (Required) : The SQS configuration. See [SQS Configuration](#sqs_configuration) for details.
+
+* `aws` (Optional) : AWS configurations. See [AWS Configuration](#aws_configuration) for details.
+
+* `on_error` (Optional) : Determines how to handle errors in SQS. Can be either `retain_messages` or `delete_messages`. If `retain_message`, then Data Prepper will leave the message in the SQS queue and try again. This is recommended for dead-letter queues. If `delete-messages`, then Data Prepper will delete failed messages. Defaults to `retain_messages`.
+
+* `buffer_timeout` (Optional) : Duration - The timeout for writing events to the Data Prepper buffer. Any Events which the S3 Source cannot write to the Buffer in this time will be discarded. Defaults to 10 seconds.
+
+* `records_to_accumulate` (Optional) : The number of messages to write to accumulate before writing to the Buffer. Defaults to 100.
+
+* `disable_bucket_ownership_validation` (Optional) : Boolean - If set to true, then the S3 Source will not attempt to validate that the bucket is owned by the expected account. The only expected account is the same account which owns the SQS queue. Defaults to `false`.
+
+### <a name="sqs_configuration">SQS Configuration</a>
+
+* `queue_url` (Required) : The SQS queue URL of the queue to read from.
+* `maximum_messages` (Optional) : Duration - The maximum number of messages to read from the queue in any request to the SQS queue. Defaults to 10.
+* `visibility_timeout` (Optional) : Duration - The visibility timeout to apply to messages read from the SQS queue. This should be set to the amount of time that Data Prepper may take to read all the S3 objects in a batch. Defaults to 30 seconds.
+* `wait_time` (Optional) : Duration - The time to wait for long-polling on the SQS API. Defaults to 20 seconds.
+* `poll_delay` (Optional) : Duration - A delay to place between reading and processing a batch of SQS messages and making a subsequent request. Defaults to 0 seconds.
+
+### <a name="aws_configuration">AWS Configuration</a>
+
+The AWS configuration is the same for both SQS and S3.
```

Comment:
Do we need to mention how AWS credentials will be fetched if STS role ARN is not used? Maybe a reference to AWS documentation.
https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html